### PR TITLE
Check for null event in Viewport::push_input

### LINKED ISSUE #77030: Executing `AcceptDialog.push_input` crashes Godot
### Godot version

4.1.dev.custom_build. c64afeb01

### System information

Ubuntu 23.04 - AMD RX 470, Gnome shell 43 X11

### Issue description

When executing
```
extends Node
func _process(delta):
	var temp_variable144 = AcceptDialog.new()
	add_child(temp_variable144)
	temp_variable144.set_process_input(true)
	temp_variable144.push_input(null, true)
```
Godot crashes:
```
scene/main/node.cpp:3070:25: runtime error: member call on null pointer of type 'const struct InputEvent'
scene/main/node.cpp:3070:25: runtime error: member access within null pointer of type 'const struct InputEvent'

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.1.dev.custom_build (c64afeb01731c477813d959daee99a99612d8077)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] godot4s() [0x2a228f2] (/home/runner/work/GodotBuilds/GodotBuilds/godot/platform/linuxbsd/crash_handler_linuxbsd.cpp:57)
[2] /lib/x86_64-linux-gnu/libc.so.6(+0x3c4b0) [0x7f537ae3c4b0] (??:0)
[3] Node::_call_input(Ref<InputEvent> const&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/node.cpp:3070)
[4] SceneTree::_call_input_pause(StringName const&, SceneTree::CallInputType, Ref<InputEvent> const&, Viewport*) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/scene_tree.cpp:1193)
[5] Viewport::push_input(Ref<InputEvent> const&, bool) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/viewport.cpp:2943)
[6] void call_with_variant_args_helper<__UnexistingClass, Ref<InputEvent> const&, bool, 0ul, 1ul>(__UnexistingClass*, void (__UnexistingClass::*)(Ref<InputEvent> const&, bool), Variant const**, Callable::CallError&, IndexSequence<0ul, 1ul>) (/home/runner/work/GodotBuilds/GodotBuilds/godot/./core/variant/binder_common.h:303 (discriminator 6))
[7] void call_with_variant_args_dv<__UnexistingClass, Ref<InputEvent> const&, bool>(__UnexistingClass*, void (__UnexistingClass::*)(Ref<InputEvent> const&, bool), Variant const**, int, Callable::CallError&, Vector<Variant> const&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/./core/variant/binder_common.h:421)
[8] MethodBindT<Ref<InputEvent> const&, bool>::call(Object*, Variant const**, int, Callable::CallError&) const (/home/runner/work/GodotBuilds/GodotBuilds/godot/./core/object/method_bind.h:331)
[9] Object::callp(StringName const&, Variant const**, int, Callable::CallError&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/core/object/object.cpp:739 (discriminator 1))
[10] Variant::callp(StringName const&, Variant const**, int, Variant&, Callable::CallError&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/core/variant/variant_call.cpp:1174)
[11] GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Callable::CallError&, GDScriptFunction::CallState*) (/home/runner/work/GodotBuilds/GodotBuilds/godot/modules/gdscript/gdscript_vm.cpp:1663)
[12] GDScriptInstance::callp(StringName const&, Variant const**, int, Callable::CallError&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/modules/gdscript/gdscript.cpp:1787)
[13] bool Node::_gdvirtual__process_call<false>(double) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/node.h:298 (discriminator 5))
[14] Node::_notification(int) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/node.cpp:62)
[15] Node::_notificationv(int, bool) (/home/runner/work/GodotBuilds/GodotBuilds/godot/./scene/main/node.h:46 (discriminator 14))
[16] Object::notification(int, bool) (/home/runner/work/GodotBuilds/GodotBuilds/godot/core/object/object.cpp:798)
[17] SceneTree::_process_group(SceneTree::ProcessGroup*, bool) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/scene_tree.cpp:944)
[18] SceneTree::_process(bool) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/scene_tree.cpp:1017 (discriminator 2))
[19] SceneTree::process(double) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/scene_tree.cpp:506 (discriminator 4))
[20] Main::iteration() (/home/runner/work/GodotBuilds/GodotBuilds/godot/main/main.cpp:3334)
[21] OS_LinuxBSD::run() (/home/runner/work/GodotBuilds/GodotBuilds/godot/platform/linuxbsd/os_linuxbsd.cpp:899)
[22] godot4s(main+0x583) [0x2a222a9] (/home/runner/work/GodotBuilds/GodotBuilds/godot/platform/linuxbsd/godot_linuxbsd.cpp:75)
[23] /lib/x86_64-linux-gnu/libc.so.6(+0x23a90) [0x7f537ae23a90] (??:0)
[24] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x89) [0x7f537ae23b49] (??:0)
[25] godot4s(_start+0x25) [0x2a21c65] (??:?)

```

This example was found by Godot fuzzer - [Qarminer](https://github.com/qarmin/Qarminer), so it is quite unlikelly that this code could be used in real project, but still this should be handled gracefully.

Memory leaks or asan backtraces are visibe when using Godot build with sanitizers support - https://github.com/qarmin/GodotBuilds/actions (linux -> linux-editor-sanitizers)

### Steps to reproduce

Above

### Minimal reproduction project

Above

### PR DESCRIPTION
**Pull request by brno32**
Fixes https://github.com/godotengine/godot/issues/77030

### CODE REVIEW

**Comment by AThousandShips** [comment]
Can you check if this is the cause of #77057 as well?

**Comment by brno32** [comment]
> Can you check if this is the cause of #77057 as well?

They're definitely related. The issue surfaces at the same place, and it's due to a null event propagating from the input method to the lower level handling, which these classes share.

I'm not sure if it's better to guard each individual method with a null check, live I've done in this PR, or to handle it here
https://github.com/godotengine/godot/blob/fb10f45efe52e330496e325b6b7efd0e52f769b0/scene/main/node.cpp#L3070

File: scene/main/viewport.cpp [hunk 1]
```diff
@@ -2904,6 +2904,7 @@ bool Viewport::_sub_windows_forward_input(const Ref<InputEvent> &p_event) {
 
 void Viewport::push_input(const Ref<InputEvent> &p_event, bool p_local_coords) {
 	ERR_FAIL_COND(!is_inside_tree());
+	ERR_FAIL_NULL_MSG(p_event, "p_event cannot be null.");
```
Comment (akien-mga) [inline scene/main/viewport.cpp:2907]: That error message isn't much more explicit than the default error message for `ERR_FAIL_NULL`, so I would just do `ERR_FAIL_NULL(p_event);`.

Comment (kleonc) [reply scene/main/viewport.cpp:2907]: Shouldn't it be either `ERR_FAIL_COND(p_event.is_null());` or `ERR_FAIL_NULL(p_event.ptr());` instead? `p_event` is a reference.

Comment (akien-mga) [reply scene/main/viewport.cpp:2907]: Indeed, good catch. `ERR_FAIL_COND(p_event.is_null());` would be the common way we check this.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.0.4.

**Pull request merged**